src/Interpreters/ChineseTokenizer.cpp

```diff
@@ -0,0 +1,38 @@
+#include "ChineseTokenizer.h"
+
+#if USE_CPPJIEBA
+
+#include <cppjieba/Jieba.hpp>
+
+namespace DB
+{
+
+ChineseTokenizer & ChineseTokenizer::instance()
+{
+    static ChineseTokenizer tokenizer;
+    return tokenizer;
+}
+
+std::vector<std::string> ChineseTokenizer::tokenize(const std::string & str, ChineseTokenizationGranularity granularity)
+{
+    std::vector<std::string> tokens;
+    switch (granularity)
```

Comment:
Note: enums are ideally always iterated using switch statements. That way, the compiler will complain if someone adds a new enum value but forgets to add a new path. Such things will go unnoticed for if/else statements.